assets/js/components/Link.js

```diff
@@ -40,13 +41,17 @@ function Link( {
 	caps,
 	danger,
 	disabled,
+	to,
 	'aria-label': ariaLabel,
 	...extraProps
 } ) {
 	// Note: the disabled attribute does not alter behavior of anchor tags,
 	// so if disabled we force it to be a button.
 	const isAnchor = href && ! disabled;
-	const SemanticLink = isAnchor ? 'a' : 'button';
+	const element = isAnchor ? 'a' : 'button';
+	// If `to` prop is set, a `react-router` link is expected
+	const isRouterLink = typeof to !== 'undefined';
+	const SemanticLink = isRouterLink ? RouterLink : element;
```

Comment:
This all makes sense but it's a bit tough, to me, to follow the nested ternary logic here. Some if/else statements might make the conditions clearer:

```suggestion
	let LinkElement;
	if ( typeof to !== 'undefined' ) {
		// If `to` prop is set, a `react-router` link is expected.
		LinkElement = RouterLink;
	} else if ( href && ! disabled ) {
		// If href` prop is set this is a "regular" link, but only render an
		// `<a>` tag if the link isn't disabled. Disabled links should be
		// rendered as `<button>`s instead.
		LinkElement = 'a';
	} else {
		LinkElement = 'button';
	}
```